source/mig.md

```diff
@@ -0,0 +1,76 @@
+# Multi-Instance GPU (MIG)
+
+Multi-Instance GPU is a technology that allows partitioning a single GPU into multiple instances, making each one seem as a completely independent GPU. Each instance then receives a certain slice of the GPU computational resources and a pre-defined block of memory that is detached from the other instances by on-chip protections.
+
+Due to the protection layer to make MIG secure, certain limitations exist. One such limitation that is generally important for HPC applications is the lack of support for CUDA Inter-Process Communication (IPC), which enables transfers over NVLink and NVSwitch to greatly speed up communication between physical GPUs. When using MIG, NVLink and NVSwitch are thus completely unavailable, forcing the application to take a more expensive communication channel via the system (CPU) memory.
+
+Given limitations in communication capability, we advise users to first understand the tradeoffs that have to be made when attempting to setup a cluster of MIG instances. While the partitioning could be beneficial to certain applications that need only a certain amount of compute capability, communication bottlenecks may be a problem and thus need to be thought of carefully.
+
+## Dask Cluster
+
+Dask clusters of MIG instances are supported via Dask-CUDA as long as all MIG instances are identical with respect to memory. Much like a cluster of physical GPUs, mixing GPUs with different memory sizes is generally not a good idea as Dask may not be able to balance work correctly and eventually could lead to more frequent out-of-memory errors.
+
+For example, partitioning two GPUs into 7 x 10GB instances each and setting up a cluster with all 14 instances should be ok. However, partitioning one of the GPUs into 7 x 10GB instances and another with 3 x 20GB should be avoided.
+
+Unlike for a system composed of unpartitioned GPUs, Dask-CUDA cannot automatically infer the GPUs to be utilized for the cluster. In a MIG setup, the user is then required to specify the GPU instances to be used by the cluster. This is achieved by specifying either the `CUDA_VISIBLE_DEVICES` environment variable for either `LocalCUDACluster` or `dask-cuda-worker`, or the homonymous argument for `LocalCUDACluster`.
```

Comment:
Maybe we should use intersphinx to link `LocalCUDACluster` to the reference in the `dask-cuda` docs?

Reply:
Sorry for my ignorance of Sphinx, I see https://github.com/rapidsai/deployment/blob/82e2bb2cf7ede6592449d444faf6d5550818d40a/source/conf.py#L71-L76 but I don't really know how to refer to those, would we need a specific entry for `LocalCUDACluster` there, another for `dask-cuda-worker`, and so on? Or do you mean to simply add a regular link?

Reply:
No worries! There are neat tricks you can do to cross-link classes and functions from other Sphinx docs. I've added the mapping and updated the mentions of `LocalCUDACluster` to map over in [82ae5c5](https://github.com/rapidsai/deployment/pull/63/commits/82ae5c5b7e5ee97aae1052be889e1851b6832d48)